sdk/identity/azure-identity/src/default_azure_credential.cpp

```diff
@@ -81,24 +92,20 @@ DefaultAzureCredential::DefaultAzureCredential(
         CredentialInfo{
             true,
             "ManagedIdentityCredential",
-            [](auto options) { return std::make_shared<ManagedIdentityCredential>(options); }},
+            [&](auto options) {
+              // If specifically 'ManagedIdentityCredential' is used, do not perform a probe
+              // request, going for the full retry with exponential backoffs instead.
+              ManagedIdentityCredentialOptions managedIdentityCredentialOptions;
+              static_cast<Core::Credentials::TokenCredentialOptions&>(managedIdentityCredentialOptions) = options;
```

Comment:
We do use this pattern, i.e. https://github.com/Azure/azure-sdk-for-cpp/blob/313a43af4f67b72914228699482931fe0be4b717/sdk/identity/azure-identity/src/environment_credential.cpp#L62-L64 and https://github.com/Azure/azure-sdk-for-cpp/blob/313a43af4f67b72914228699482931fe0be4b717/sdk/identity/azure-identity/src/workload_identity_credential.cpp#L37-L43

Customers won't need to do this, it is only us who does this in credentials that wrap other credentials. I don't think we should add a constructor for an option struct for his purpose, and add API. static_cast wotrks just fine. It is legit C++.